README.md

```diff
@@ -1,60 +1,128 @@
-RedshiftInputFormat
-===
+# RedshiftInputFormat
 
 [![Build Status](https://travis-ci.org/databricks/spark-redshift.svg?branch=master)](https://travis-ci.org/databricks/spark-redshift)
 [![codecov.io](http://codecov.io/github/databricks/spark-redshift/coverage.svg?branch=master)](http://codecov.io/github/databricks/spark-redshift?branch=master)
 
-Install
--------
+A library to load data into Spark SQL DataFrames from Amazon Redshift, and write them back to
+Redshift tables. Amazon S3 is used to transfer data efficiently into and out of Redshift, and
+JDBC is used to trigger the appropriate <tt>COPY</tt> and <tt>UNLOAD</tt> commands on Redshift automatically.
 
-You may use this library in your applications through:
+## Install
 
-### spark-shell, pyspark, or spark-submit (with Spark 1.3)
+**Note:** `spark-redshift` requires Apache Spark version 1.4+ and Amazon Redshift version 1.0.963+ for
+writing with Avro data.
+
+You may use this library in your applications with the following dependency information:
 
 ```
-> $SPARK_HOME/bin/spark-shell --packages databricks:spark-redshift:0.3
+groupId: com.databricks
+artifactId: spark-redshift
+version: 0.4.1
 ```
 
-### sbt
+The project makes use of [`spark-avro`](https://github.com/databricks/spark-avro), which is pulled
+in as a dependency, however you'll need to provide the corresponding `avro-mapred` matching the Hadoop
+distribution that you plan to deploy to.
 
-If you use the sbt-spark-package plugin, in your sbt build file, add:
+Further, as Redshift is an AWS product, some AWS libraries will be required. This library expects that
+your deployment environment will include `hadoop-aws`, or other things necessary to access S3, credentials,
+etc. Check the dependencies with "provided" scope in <tt>build.sbt</tt> if you're at all unclear.
 
-```
-spDependencies += "databricks/spark-redshift:0.3"
-```
+You're also going to need a JDBC driver that is compatible with Redshift. The one used for testing can be
+found in <tt>build.sbt</tt>, however Amazon recommend that you use [their driver](http://docs.aws.amazon.com/redshift/latest/mgmt/configure-jdbc-connection.html).
+
+## Usage
+
+### Data Sources API
+
+You can use `spark-redshift` via the Data Sources API in Scala, Python or SQL, as follows:
+
+#### Scala
+
+```scala
+import org.apache.spark.sql._
+
+val sc = // existing SparkContext
+val sqlContext = new SQLContext(sc)
 
-Otherwise,
+// These three settings are required, see below for full information
+val params = Map("jdbcurl" -> "jdbc:postgresql://redshifthost:5439/database?user=username&password=pass",
+                 "redshifttable" -> "my_table",
+                 "tempdir" -> "s3://path/for/temp/data")
+
+// Get some data from a Redshift table
+val df: DataFrame = sqlContext.read
+    .format("com.databricks.spark.redshift")
+    .options(params)
```

Comment:
I'd consider using the following syntax (in general I try to use syntax that works across languages wherever possible in example code)

```
.option("url", "jdbc:postgresql://redshifthost:5439/database?user=username&password=pass")
.option("table", "my_table")
```